indico/modules/categories/controllers/display.py

```diff
@@ -384,3 +388,171 @@ def _checkParams(self):
+        start_dt = self.start_dt - relativedelta(days=self.start_dt.weekday())
+        end_dt = self.end_dt + relativedelta(days=(7 - self.end_dt.weekday()) % 7)
+        return self._get_days(start_dt, end_dt)
+
+    @staticmethod
+    def _get_days(start_dt, end_dt):
+        # Return all days in the open-ended interval
+        current_dt = start_dt
+        while current_dt < end_dt:
+            yield current_dt
+            current_dt = current_dt + relativedelta(days=1)
+
+    @staticmethod
+    def _pop_head_while(predicate, list_):
+        # Pop the head of the list while the predicate is true and return the popped elements
+        res = []
+        while len(list_) and predicate(list_[0]):
+            res.append(list_[0])
+            list_.pop(0)
+        return res
+
+    def _other_day_url(self, date):
+        return url_for('.overview', self.category, detail=self.detail, period=self.period,
+                       date=format_date(date, 'YYYY-MM-dd'))
+
+    def _process_multiday_events(self, info, event):
+        # Add "fake" proxy events for events spanning multiple days such that there is one event per day
+        # Function type: Event -> List[Event]
+        tzinfo = self.category.display_tzinfo
+
+        class _EventProxy(object):
+            def __init__(self, event, date, timetable_objects):
+                start_dt = datetime.combine(date, event.start_dt.astimezone(tzinfo).timetz())
+                assert date >= event.start_dt
+                assert date <= event.end_dt
+                object.__setattr__(self, '_start_dt', start_dt)
+                object.__setattr__(self, '_real_event', event)
+                object.__setattr__(self, '_timetable_objects', timetable_objects)
+
+            def __getattribute__(self, name):
+                if name == 'start_dt':
+                    return object.__getattribute__(self, '_start_dt')
+                event = object.__getattribute__(self, '_real_event')
+                if name == 'timetable_objects':
+                    return object.__getattribute__(self, '_timetable_objects')
+                if name == 'ongoing':
+                    return event.start_dt.date() != self.start_dt.date()
+                if name == 'first_occurence_start_dt':
+                    return event.start_dt
+                return getattr(event, name)
+
+            def __setattr__(self, name, value):
+                raise AttributeError('This instance is read-only')
+
+            def __repr__(self):
+                return '<_EventProxy({}, {})>'.format(self.start_dt, object.__getattribute__(self, '_real_event'))
+
+        # Breaks, contributions and sessions grouped by start_dt. Each EventProxy will return the relevant ones only
+        timetable_objects = sorted(chain(*info[event.id].values()), key=attrgetter('timetable_entry.start_dt'))
+        def _key(timetable_object):
+            return timetable_object.timetable_entry.start_dt.astimezone(tzinfo).date()
```

Comment:
All timetable-associated objects (breaks/contributions/session-blocks) have a `start_dt` property that returns `self.timetable_entry.start_dt` so you can shorten this a bit and it'd probably even fit nicely in a lambda